include/dlaf/eigensolver/band_to_tridiag.h

```diff
@@ -89,5 +89,32 @@ TridiagResult<T, Device::CPU> bandToTridiag(blas::Uplo uplo, SizeType band_size,
 
   return DLAF_UNREACHABLE(TridiagResult<T, Device::CPU>);
 }
+
+template <Backend backend, Device device, class T>
+TridiagResult<T, Device::CPU> bandToTridiag(comm::CommunicatorGrid grid, blas::Uplo uplo,
+                                            SizeType band_size, Matrix<const T, device>& mat_a) {
+  DLAF_ASSERT(matrix::square_size(mat_a), mat_a);
+  DLAF_ASSERT(matrix::square_blocksize(mat_a), mat_a);
+  DLAF_ASSERT(matrix::equal_process_grid(mat_a, grid), mat_a, grid);
+
+  // If the grid contains only one rank force local implementation.
+  if (grid.size() == comm::Size2D(1, 1))
+    return bandToTridiag<backend, device, T>(uplo, band_size, mat_a);
```

Comment:
Nothing to do for this comment, but I'm just curious: is this a shortcut we can apply for other algorithms as well or is there something special about this one that allows it?

Reply:
We can do it for all other algorithms. What is special about this one? The fact that the distributed is implemented assuming at least two ranks, so it will not work without it.